src/commonMain/kotlin/operations/numeric/ComparingOperation.kt

```diff
@@ -0,0 +1,67 @@
+package operations.numeric
+
+import comparableList
+import secondOrNull
+
+internal interface ComparingOperation {
+    fun compareListOfTwo(values: List<Any?>?, operator: (Int, Int) -> Boolean) = values?.comparableList
+        ?.takeIf { it.size >= 2 }
+        ?.compare(operator) ?: false
+
+    fun compareOrBetween(
+        values: List<Any?>?,
+        operator: (Int, Int) -> Boolean
+    ) = values?.comparableList?.let { comparableList ->
+        when {
+            comparableList.size == 2 -> comparableList.compare(operator)
+            comparableList.size > 2 -> comparableList.between(operator)
+            else -> false
+        }
+    } ?: false
+
+    private fun List<Comparable<*>?>.compare(operator: (Int, Int) -> Boolean): Boolean {
+        return compareOrNull(firstOrNull(), secondOrNull())?.let { operator(it, 0) } ?: false
+    }
+
+    private fun List<Comparable<*>?>.between(operator: (Int, Int) -> Boolean): Boolean {
+        val firstEvaluation = compareOrNull(firstOrNull(), secondOrNull())
+        val secondEvaluation = compareOrNull(secondOrNull(), getOrNull(2))
```

Comment:
there is only one call for that getOrNull(2), maybe thirdOrNull? Just to be in line with the rest. Not blocker.

Reply:
Dont want to do it in advance. Soon I will be working on comparison operators and will probably extract it to such function